src/goto-programs/abstract-interpretation/interval_domain.cpp

```diff
@@ -58,6 +58,27 @@ void interval_domaint::update_symbol_interval(
   wrap_map[sym.thename] = value;
 }
 
+template <class Interval>
+void interval_domaint::apply_assume_symbol_truth(
+  const symbol2t &sym,
+  bool is_false)
+{
+  Interval interval = get_interval_from_symbol<Interval>(sym);
+  // [0,0]
+  if(is_false)
+  {
+    interval.set_upper(0);
+    interval.set_lower(0);
+  }
+  // [1, infinity]
+  else if(
+    sym.type->type_id == type2t::unsignedbv_id ||
+    sym.type->type_id == type2t::bool_id)
```

Comment:
Is this condition possibly missing signed BVs and maybe floats as well? The new code below in `assume_rec()` seems to apply this function to symbols of those types as well and they do not seem be handled here.

Reply:
> Is this condition possibly missing signed BVs and maybe floats as well? 
Yeah, they kind are. For signedbv there are a few corner cases that could be optimized (when the interval was inside (-infinity, 0] or [0, infinity)). 

Floats I am not sure how to handle yet.

Reply:
> Floats I am not sure how to handle yet.

Maybe add a TODO here until we have an implementation. Even if you know about the issue, others reading the code might have questions similar to mine.